presto-main/src/main/java/com/facebook/presto/SessionRepresentation.java

```diff
@@ -212,7 +214,7 @@ public Session toSession(SessionPropertyManager sessionPropertyManager)
                 new QueryId(queryId),
                 transactionId,
                 clientTransactionSupport,
-                new Identity(user, Optional.empty()),
+                new Identity(user, InternalPrincipal.createPrincipal(getPrincipal())),
```

Comment:
Since it's instance method I'd suggest using `principal` field directly instead of using getter. Also, this is what the rest of the method does.